File: week 2/fibonacci/Fib.java
import java.util.Scanner;

public class Fib {
  private static long calc_fib(int n) {

	if(n <= 1)
		return n;
	else if(n == 2)
		return 1;
	else {
		int [] result = new int[n+1];
		result[1] = 1;
		result[2] = 1;
		for(int i=3;i<n+1;i++) {
			result[i] = result[i-1] + result[i-2];
		}
		return result[n];
	}
  }

  public static void main(String args[]) {
    Scanner in = new Scanner(System.in);
    int n = in.nextInt();

    System.out.println(calc_fib(n));
  }
}

File: week 2/fibonacci/fib.cpp
#include <iostream>
#include <string.h>
#include <vector>

using std::vector;

int calc_fib(int n) {
	vector<int> result(n+1);
	result[0] = 0;
	result[1] = 1;
	for(int i=2; i< result.size();i++) {
	  result[i] = result[i - 1] + result[i - 2];
	}
	return result[n];
}

int main() {
    int n = 0;
    std::cin >> n;

    std::cout << calc_fib(n) << '\n';
    return 0;
}
